### PR TITLE
Support ESLint 8.x

### LINKED ISSUE #286: Support ESLint 8.x
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

It would be awesome to have official ESLint 8 support. 👊
I'm happy to help where I can of course 🙂

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #286

### CODE REVIEW

**Comment by yassin-kammoun-sonarsource** [comment]
I don't have permission to commit changes on [release.yml](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/.github/workflows/release.yml), and here in particular:

https://github.com/SonarSource/eslint-plugin-sonarjs/blob/0331fd05b20df41b5e35601fa84fc80e17ad5fb2/.github/workflows/release.yml#L23

File: package.json [hunk 1]
```diff
@@ -30,31 +30,32 @@
     "check-format": "prettier --list-different \"{src,tests}/**/*.ts\""
   },
   "peerDependencies": {
-    "eslint": "^5.0.0 || ^6.0.0 || ^7.0.0"
+    "eslint": "^8.0.0"
```
Comment (vilchik-elena) [inline package.json:33]: why can't we keep compatibility with other older versions?

Comment (yassin-kammoun-sonarsource) [reply package.json:33]: Well, I tried using our plugin with eslint 5.16.0 and I got some type errors on linting. Not sure how I managed to do that, but I am no longer able to reproduce them. I will revert that change.

File: src/rules/cognitive-complexity.ts [hunk 2]
```diff
@@ -142,7 +146,11 @@ const rule: Rule.RuleModule<string, Options> = {
         if (isFileComplexity) {
           // as issues are the only communication channel of a rule
           // we pass data as serialized json as an issue message
-          context.report({ node, message: fileComplexity.toString() });
+          context.report({
+            node,
+            messageId: 'refactorFunction',
```
Comment (vilchik-elena) [inline src/rules/cognitive-complexity.ts:151]: we should use another message id here, to have final message to contain only `fileComplexity` value (https://github.com/SonarSource/SonarJS/blob/master/eslint-bridge/src/linter.ts#L389)

File: src/rules/cognitive-complexity.ts [hunk 3]
```diff
@@ -43,14 +42,18 @@ type LoopStatement =
...
+const rule: TSESLint.RuleModule<string, (number | 'metric' | 'sonar-runtime')[]> = {
   meta: {
+    messages: {
+      refactorFunction: message,
+      sonarRuntime: '{{sonarRuntimeData}}',
+    },
     type: 'suggestion',
     docs: {
       description: 'Cognitive Complexity of functions should not be too high',
-      category: 'Best Practices',
```
[the remark this answers is not in the document]
Comment (yassin-kammoun-sonarsource) [reply src/rules/cognitive-complexity.ts:53]: This was dropped (see https://github.com/eslint/eslint/issues/13398)

File: ruling/index.ts [hunk 4]
```diff
@@ -19,15 +19,17 @@
...
 import * as path from "path";
-import { CLIEngine } from "eslint";
+import { ESLint } from "eslint";
 import * as lodash from "lodash";
 import * as minimist from "minimist";
 
 const rulesPath = path.join(__dirname, "../lib/rules");
 
-run();
+run().catch((error) => {
```
[the remark this answers is not in the document]
Comment (yassin-kammoun-sonarsource) [reply ruling/index.ts:28]: I missed that; I would say it was added for debugging purpose. I will remove it.

**Review by vilchik-elena — APPROVED** [review]

**Comment by vilchik-elena** [comment]
@yassin-kammoun-sonarsource I am not satisfied with the way `sonar-runtime` messages are done now, but I don't find a better way so far :( 
So please have a look at the changes I did (should be minor) and I suggest we test this build in SonarJS before doing the merge of this PR

File: tests/rules/cognitive-complexity.test.ts [hunk 5]
```diff
@@ -650,7 +650,7 @@ class TopLevel {
 }
       `,
       options: [0, 'metric'],
-      errors: [complexity(25)],
+      errors: [{ messageId: 'fileComplexity', data: { complexityAmount: complexity.toString() } }],
```
Comment (yassin-kammoun-sonarsource) [inline tests/rules/cognitive-complexity.test.ts:653]: @vilchik-elena Something is fishy here: we are calling `toString` on a function symbol.

**Pull request merged**